upf/upf_cli.c

```diff
@@ -1108,34 +1121,33 @@ upf_show_session_command_fn (vlib_main_t * vm,
   u8 has_cp_f_seid = 0, has_up_seid = 0;
   upf_session_t *sess = NULL;
   int debug = 0;
-#if FLOWTABLE_TODO
+  u32 limit = MAX_SHOW_UPF_SESSIONS;
   u8 has_flows = 0;
-#endif
 
   if (unformat_user (main_input, unformat_line_input, line_input))
     {
       while (unformat_check_input (line_input) != UNFORMAT_END_OF_INPUT)
 	{
-	  if (unformat (line_input, "cp %U seid %lu",
+	  if (unformat (line_input, "cp %U seid 0x%llx",
 			unformat_ip46_address, &cp_ip, IP46_TYPE_ANY,
 			&cp_seid))
 	    has_cp_f_seid = 1;
-	  else if (unformat (line_input, "cp %U seid 0x%lx",
+	  else if (unformat (line_input, "cp %U seid %llu",
 			     unformat_ip46_address, &cp_ip, IP46_TYPE_ANY,
 			     &cp_seid))
 	    has_cp_f_seid = 1;
-	  else if (unformat (line_input, "up seid %lu", &up_seid))
+	  else if (unformat (line_input, "up seid 0x%llx", &up_seid))
 	    has_up_seid = 1;
-	  else if (unformat (line_input, "up seid 0x%lx", &up_seid))
+	  else if (unformat (line_input, "up seid %lu", &up_seid))
 	    has_up_seid = 1;
 	  else if (unformat (line_input, "debug"))
 	    debug = 1;
-#if FLOWTABLE_TODO
-	  else if (unformat (line_input, "%lu flows", &up_seid))
+	  else if (unformat (line_input, "0x%llx flows", &up_seid))
 	    has_flows = 1;
-	  else if (unformat (line_input, "0x%lx flows", &up_seid))
+	  else if (unformat (line_input, "flows", &up_seid))
 	    has_flows = 1;
-#endif
+	  else if (unformat (line_input, "limit %u", &limit))
+	    ;
```

Comment:
should we enforce a hard max for limit? Currently setting limit to 0 would effectively mean `unlimited`

Reply:
Added the hard limit of 10K sessions